### PR TITLE
# 169258627 Deployment of base application to heroku

### PR DESCRIPTION
**Pull request by ivymwende**
#### What does this PR do?
Deploys the base application on Heroku

#### Description of Task to be completed?
- Set up Procfile 
- Set up the staging environment on Heroku
- Set up the production environment on Heroku

#### How should this be manually tested?
- Download postman
- Via postman, use the link `https://team-odd-bn-backend-staging.herokuapp.com/` to check the staging environment
- Via postman, use the link `https://team-odd-barefoot-backend.herokuapp.com/` to check the production environment

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#169258627](https://www.pivotaltracker.com/story/show/169258627)

#### Screenshots (if appropriate)
Screen Shot 2019-11-08 at 15 06 13
Screen Shot 2019-11-08 at 15 23 57

### CODE REVIEW

**Comment by victkarangwa** [comment]
Hooray!
Congraturations @ivymwende for deploying the app
Screen Shot 2019-11-08 at 17 17 50

**Comment by victkarangwa** [comment]
How about we get rid of these houndCI violations?

**Review by hezronkimutai — APPROVED** [review]

**Review by victkarangwa — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -4,6 +4,8 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
+    "start": "NODE_ENV=dev nodemon ./src/index.js --exec babel-node --",
```
Comment (NiyoEric) [inline package.json:7]: @ivymwende i think `NODE_ENV=production` is clear

File: src/index.js [hunk 2]
```diff
@@ -1,93 +1,21 @@
...
-const isProduction = process.env.NODE_ENV === "production";
-
-// Create global app object
 const app = express();
+const port = process.env.PORT || 8000;
 
-app.use(cors());
-
-// Normal express config defaults
-app.use(require("morgan")("dev"));
```
Comment (NiyoEric) [inline src/index.js:21]: @ivymwende we need `morgan` package in development.

File: src/index.js [hunk 3]
```diff
@@ -1,93 +1,21 @@
...
-});
+app.get('/', (req, res) => res.status(200).send({
+  message: 'Welcome to Barefoot Nomad.'
+}));
 
-// finally, let's start our server...
-const server = app.listen(process.env.PORT || 3000, function() {
-    console.log("Listening on port " + server.address().port);
+app.listen(port, () => {
+  // eslint-disable-next-line no-console
```
Comment (NiyoEric) [inline src/index.js:18]: @ivymwende put all eslint rules such as `// eslint-disable-next-line no-console` in `.eslintrc.json`

File: src/index.js [hunk 4]
```diff
@@ -1,93 +1,21 @@
...
-    mongoose.connect("mongodb://localhost/conduit");
-    mongoose.set("debug", true);
-}
-
-require("./models/User");
-
-app.use(require("./routes"));
-
-/// catch 404 and forward to error handler
-app.use(function(req, res, next) {
```
Comment (NiyoEric) [inline src/index.js:53]: @ivymwende we need this middleware to catch 404 errors

File: package.json (hunk 1 above)
Comment (ivymwende) [reply package.json:7]: @NiyoEric What do you mean by this?

**Review by julietezekwe — APPROVED** [review]
LGTM

**Pull request merged**